hud.lua

```diff
@@ -19,6 +19,10 @@ local all_waypoint_huds = {} -- wpid -> plname -> hud_id
 
 local function show_waypoint_to_player(plname, waypoint)
 	local player = minetest.get_player_by_name(plname)
+        if not player then
+           return
+        end
+
```

Comment:
mixing different indentation styles here

There were several more places where this race condition had to be handled, I created a separate commit that fixes all this.